great_expectations/data_context/util.py

```diff
@@ -153,9 +156,14 @@ def file_relative_path(dunderfile, relative_path):
     It is meant to be used like the following:
     file_relative_path(__file__, 'path/relative/to/file')
 
+    This has been modified from Dagster's utils:
     H/T https://github.com/dagster-io/dagster/blob/8a250e9619a49e8bff8e9aa7435df89c2d2ea039/python_modules/dagster/dagster/utils/__init__.py#L34
     """
-    return os.path.join(os.path.dirname(dunderfile), relative_path)
+    abs_path = pathlib.Path(source_path).joinpath(relative_path).absolute()
+    str_path = abs_path.as_posix()
```

Comment:
Can we just return a `pathlib.Path` as well? Then this call is unnecessary.

Reply:
Why do we need to call `as_posix()` at all? Why not just `str(abs_path)`?
Does the rest of our filepath manipulation covert windows paths to be posix paths?